wagtailio/newsletter/models.py

```diff
@@ -3,39 +3,67 @@
 from django.shortcuts import render
 
 from wagtail.admin.panels import FieldPanel
-from wagtail.fields import RichTextField
+from wagtail.contrib.settings.models import BaseGenericSetting, register_setting
+from wagtail.fields import RichTextField, StreamField
 from wagtail.models import Page
 from wagtail.search import index
 
+from wagtail_newsletter.models import NewsletterPageMixin
+
+from wagtailio.newsletter.blocks import NewsletterContentBlock
 
-class NewsletterPage(Page):
-    date = models.DateField("Newsletter date")
-    intro = RichTextField(blank=True)
-    body = RichTextField()
+
+@register_setting
+class NewsletterSettings(BaseGenericSetting):
+    footer = StreamField(NewsletterContentBlock(), blank=True, use_json_field=True)
```

Comment:
@mgax this and the StreamField below don't require setting `use_json_field` anymore, see https://docs.wagtail.org/en/stable/releases/6.0.html#streamfield-no-longer-requires-use-json-field-true


```suggestion
    footer = StreamField(NewsletterContentBlock(), blank=True)
```